### PR TITLE
Fix to handle negative exptime as common function

### PR DESCRIPTION
**Pull request by minkikim89**
Every way to handle negative expiretime is different. 
I united as one.

### CODE REVIEW

**Comment by dormando** [comment]
Thanks! Unfortunately I just modified some of this same code in another branch that I just merged. You'll have to update before I can review. It might be better as a macro since it can be a hot spot, too.

**Comment by minkikim89** [comment]
@dormando 
I fixed. Please review it

File: memcached.c [hunk 1]
```diff
@@ -3911,7 +3918,7 @@ static inline void process_get_command(conn *c, token_t *tokens, size_t ntokens,
                 goto stop;
             }
 
-            it = limited_get(key, nkey, c, exptime, should_touch, DO_UPDATE, &overflow);
+            it = limited_get(key, nkey, c, realtime(exptime), should_touch, DO_UPDATE, &overflow);
```
Comment (dormando) [inline memcached.c]: move the realtime() conversion back to the if (should_touch) section. no reason to keep converting it.

File: memcached.c [hunk 2]
```diff
@@ -4152,10 +4160,8 @@ static int _meta_flag_preparse(token_t *tokens, const size_t ntokens,
                     *errstr = "CLIENT_ERROR bad token in command line format";
                     of->has_error = 1;
                 } else {
-                    if (tmp_int < 0) {
-                        tmp_int = REALTIME_MAXDELTA + 1;
-                    }
-                    of->autoviv_exptime = realtime(tmp_int);
+                    tmp_time = EXPTIME_TO_POSITIVE_TIME(tmp_int);
+                    of->autoviv_exptime = realtime(tmp_time);
```
Comment (dormando) [inline memcached.c]: of->autoviv_exptime = realtime(EXPTIME_TO_POSITIVE_TIME(tmp_time)) ?

**Comment by minkikim89** [comment]
@dormando 
I fixed it. And I added test case.

**Comment by dormando** [comment]
Looks like the tests are failing on travis. Did you run them locally?

**Comment by minkikim89** [comment]
@dormando 
I made a mistake. It works well now.

**Comment by dormando** [comment]
merged into next, thanks!

**Comment by dormando** [comment]
Fwiw though, I would accept even more tests (make another PR): you added a test for the `set` code path, which was functionally equivalent here (good to test), you also changed touch and gat so those should have specific tests so you can see the before/after impact.

**Comment by minkikim89** [comment]
OK. I'll do it later

**Comment by dormando** [comment]
out in 1.6.1!

**Pull request closed without merging**